### PR TITLE
Helm chart for app.

### PR DESCRIPTION
**Pull request by igor-kostin-1**

### CODE REVIEW

File: infrastructure/module/eks/sa.helm_release.tf [hunk 1]
```diff
@@ -35,4 +35,14 @@ resource "helm_release" "alb-controller" {
     name  = "serviceAccount.annotations.eks\\.amazonaws\\.com/role-arn"
     value = module.lb_role.iam_role_arn
   }
+
```
Comment (Igor-Knezevic-VegaIT) [inline infrastructure/module/eks/sa.helm_release.tf]: You don't need this at all, both webhook and dnsDomain.

File: kube/helm/app/templates/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ .Chart.Name }}
+  namespace: vegait-training
+  annotations:
+    kubernetes.io/ingress.class: alb
+    alb.ingress.kubernetes.io/scheme: internet-facing
+    alb.ingress.kubernetes.io/target-type: ip
+    alb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:eu-central-1:162340708442:certificate/7295afe7-3e51-4271-89eb-4d7bd8192571
```
Comment (Igor-Knezevic-VegaIT) [inline kube/helm/app/templates/ingress.yaml]: You don't need this, LB will find cert provisioned via ACM automatically based on the host that you specified here in ingress.

File: kube/helm/app/templates/secret.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: db-secret
+  namespace: vegait-training
+type: Opaque
+data:
+  POSTGRES_PASSWORD: {{ .Values.postgresql.password | b64enc }}
```
Comment (Igor-Knezevic-VegaIT) [inline kube/helm/app/templates/secret.yaml]: Why do you do base64 ?

File: infrastructure/module/eks/sa.helm_release.tf (hunk 1 above)
Comment (igor-kostin-1) [reply infrastructure/module/eks/sa.helm_release.tf]: Have we achieved that by applying `attach_load_balancer_controller_policy = true` in the Load balancer role creation file `sa.lb_role.tf`?

File: kube/helm/app/templates/secret.yaml (hunk 3 above)
Comment (igor-kostin-1) [reply kube/helm/app/templates/secret.yaml]: Without that, I had a problem in that my app failed to authenticate to db and did not make the password visible as plain text(a bit obfuscating).

Comment (Igor-Knezevic-VegaIT) [reply kube/helm/app/templates/secret.yaml]: Something else was the issue there, you don't need it as each secret by default ends up base64 encoded by K8S.

File: infrastructure/module/dns/data.tf [hunk 4]
```diff
@@ -0,0 +1,15 @@
+
+data "aws_acm_certificate" "certificate" {
```
Comment (Igor-Knezevic-VegaIT) [inline infrastructure/module/dns/data.tf]: This is not needed, you have to use ACM module to create certificate.

File: infrastructure/module/eks/sa.helm_release.tf (hunk 1 above)
Comment (Igor-Knezevic-VegaIT) [reply infrastructure/module/eks/sa.helm_release.tf]: No, this is webhook for mutation of service type and we will not utilise that feature, check docs.

Comment (Igor-Knezevic-VegaIT) [reply infrastructure/module/eks/sa.helm_release.tf]: Just to clarify, I am talking about this added stuff, not about: **serviceAccount.annotations.eks\\.amazonaws\\.com/role-arn**

File: infrastructure/module/eks/sa.helm_release.tf [hunk 5]
```diff
@@ -32,7 +32,7 @@ resource "helm_release" "alb-controller" {
   }
 
   set {
-    name  = "serviceAccount.annotations.eks\\.amazonaws\\.com/role-arn"
```
Comment (Igor-Knezevic-VegaIT) [inline infrastructure/module/eks/sa.helm_release.tf]: You need role annotation but not webhookTLS.
Also check to add **defaultTargetType** to be **IP**.

**Review by Igor-Knezevic-VegaIT — APPROVED** [review]

**Pull request merged**